Phase 2

chloebercy opened the pull request:
notes
-
- fix fat index (divide by 2048 find remainder)
- fs_create() error handling

resources
-
- [5.7 String/Array Comparison](https://www.gnu.org/software/libc/manual/html_mono/libc.html#String_002fArray-Comparison)
- [The stat structure](https://man7.org/linux/man-pages/man2/lstat.2.html)
- [fstat(2) - Linux man page](https://linux.die.net/man/2/fstat)


concerns
-
ongoing from phase 1
- define structs later for easier reading?
- error checking in order to verify that the file system has the expected format? _(anything besides: "signatures, total amounts of blocks, root directory")_

cxbercy committed:
working p2 funcs + phase 2 simple tester script

jordlim committed:
P2 functions mostly done BUT not tested, FD struct started

jordlim commented:
### notes
- fs_create() and fs_delete() error handling done minus the FD parts from phase 3 + questions below
- I kept mounted for create and delete checking
- changed rd to make it simpler, allocating may not be necessary bc it's a basic array
- started file_descriptor struct. unsure if we need the packed attribute for this at all? I don't think we do

### concerns
- may need to test filename length check in fs_create(), but I think I accounted for null char correctly
- what do we check for if filename is invalid? null chars?
- have not tested functions, also need to test the error messages on reference
- fs_create() line 214: added block_write, unsure of if we need things like this here or what it's doing entirely so I commented it out lol. functions from disk.[ch] are a lil confusing but I will spend some time to understand those better
- for fs_delete(), do we want to rename 'dataIndex'? reworked 'dataIndexNext' a bit, and we have 'entry' as well
- piazza post about FAT block, ours should work but good to consider [link](https://piazza.com/class/lcjq5p6ss7z1n0/post/199)

cxbercy committed:
error handling to fs_create() and clean up fat entry deletion in fs_delete()

chloebercy commented:
notes
-
- file_create() now checks for null filename strings (check against "/0" so null character)
- simplified while loop from line 244 and changed var names
- I don't think we don't need to update disk after file creation, because from what I understand- we manipulate the virtual file system for a bit and then finally update the disk after we unmount and like "submit the changes" almost.
